src/components/survey.js

```diff
@@ -14,12 +14,14 @@ module.exports = function(app) {
           angular.forEach(component.values, function(v) {
             values[v.value] = v.label;
           });
-          angular.forEach(component.questions, function(question) {
-            view += '<tr>';
-            view += '<th>' + question.label + '</th>';
-            view += '<td>' + values[data[question.value]] + '</td>';
-            view += '</tr>';
-          });
+		  if(data) {
```

Comment:
I am getting 'data' to be undefined when it reads it sometimes and the 'if' statement does the work. i think this fixes it if i am wrong please do let me know.